server/README.md

```diff
@@ -0,0 +1,139 @@
+# Language server bitbkae
+
+Language server for bitbake.
+
+![screenshot](https://github.com/yoctoproject/vscode-bitbake/assets/32936898/82e16428-e5fb-4e37-a2fa-b56c00cc4f4c)
+
+## Install
+
+```sh
+npm install -g language-server-bitbake
+```
+
+## Configure
+
+- For windows, change `~/.config` to `~/AppData/Local`
+- For macOS, change `~/.config` to `~/Library`
+
+### (Neo)[Vim](https://www.vim.org)
+
+For vim:
+
+- Change `~/.config/nvim` to `~/.vim`
+- Change `init.vim` to `vimrc`
+
+#### [coc.nvim](https://github.com/neoclide/coc.nvim)
+
+`~/.config/nvim/coc-settings.json`:
+
+```json
+{
+  "languageserver": {
+    "bitbake": {
+      "command": "language-server-bitbake",
+      "args": ["--stdio"],
+      "filetypes": [
+        "bitbake"
+      ]
+    }
+  }
+}
+```
+
+#### [vim-lsp](https://github.com/prabirshrestha/vim-lsp)
+
+`~/.config/nvim/init.vim`:
+
+```vim
+if executable('language-server-bitbake')
+  augroup lsp
+    autocmd!
+    autocmd User lsp_setup call lsp#register_server({
+          \ 'name': 'bitbake',
+          \ 'cmd': {server_info->['language-server-bitbake', '--stdio']},
+          \ 'whitelist': ['bitbake'],
+          \ })
+  augroup END
+endif
+```
+
+### [Neovim](https://neovim.io)
+
+`~/.config/nvim/init.lua`:
+
+```lua
+vim.api.nvim_create_autocmd({ "BufEnter" }, {
+  pattern = { "bitbake*" },
+  callback = function()
+    vim.lsp.start({
+      name = "bitbake",
+      cmd = { "language-server-bitbake", "--stdio" }
+    })
+  end,
+})
+```
+
+### [Emacs](https://www.gnu.org/software/emacs)
+
+`~/.emacs.d/init.el`:
+
+```lisp
+(make-lsp-client :new-connection
+(lsp-stdio-connection
+  `(,(executable-find "language-server-bitbake" "--stdio")))
+  :activation-fn (lsp-activate-on "*.bb")
+  :server-id "bitbake")))
+```
+
+### [Helix](https://helix-editor.com/)
+
+`~/.config/helix/languages.toml`:
+
+```toml
+[[language]]
+name = "bitbake"
+language-servers = [ "language-server-bitbake",]
+
+[language_server.language-server-bitbake]
+command = "language-server-bitbake --stdio"
+```
+
+### [KaKoune](https://kakoune.org/)
+
+#### [kak-lsp](https://github.com/kak-lsp/kak-lsp)
+
+`~/.config/kak-lsp/kak-lsp.toml`:
+
+```toml
+[language_server.language-server-bitbake]
+filetypes = [ "bitbake",]
+command = "language-server-bitbake --stdio"
+```
+
+### [Sublime](https://www.sublimetext.com)
+
+`~/.config/sublime-text-3/Packages/Preferences.sublime-settings`:
+
+```json
+{
+  "clients": {
+    "bitbake": {
+      "command": [
+        "language-server-bitbake",
+        "--stdio"
+      ],
+      "enabled": true,
+      "selector": "source.bitbake"
+    }
+  }
+}
+```
+
+### [Visual Studio Code](https://code.visualstudio.com/)
+
+[vscode-bitbake](https://github.com/yoctoproject/vscode-bitbake)
+
+## Related projects
+
+- [bitbake-language-server](https://github.com/Freed-Wu/bitbake-language-server/):
```

Comment:
Your project looks great! We do provide definitions and diagnostics in our language server but it may require our client and some VSCode specific things to work. So far both your server and what you could make work from ours on VIM seem complimentary. What's your plan for the VIM experience? Should vim users currently install both? Do you want to import the missing features from ours into yours? The other way around?

Reply:
In fact Install many language servers for one language is normal, e.g., I use `ruff-lsp` and `pyright` for python, and I think use two language servers will not be a problem for users.

> import the missing features from ours into yours

I don't know what feature is missing because I haven't used vscode-bitbake. Some difference is vscode-bitbake can bundle a document however I expect user have docuement in their projects. BTW, `scripts/fetch-docs.sh` is very very slow. I think it is not a good develop experience to retry many times to make `scripts/fetch-docs.sh` work.

Reply:
Oh, I run off topic.

> Do you want

I must use vscode-bitbake (language-server-bitbake) for a long time to know what feautures is missing and which features can be imported. Whatever, it should be a future work.

> install both

I level the decision permission to users :smile:

>  plan for the VIM experience

In fact, I think LSP should be a cross-editor protocol, So it is not `VIM experience`, should be all editor's experience :smile:

Reply:
> In fact, I think LSP should be a cross-editor protocol, So it is not `VIM experience`, should be all editor's experience 😄

You are definetely right! However, our current LSP is very tied in to the client and VSCode to get all the features working. Without lots of human resources to refactor it and maintain that in the future, I don't believe we could get all those features to work cross-editor. I'm really glad with the limited port you are providing! It's a good start and remains compatible with VSCode.